src/app/components/book-dialog/add/add-dialog.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, Inject, ChangeDetectionStrategy, EventEmitter } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { ReadBook } from '@app/shared/models/read-book.model';
+import { FormActions } from '@app/shared/enums';
+
+@Component({
+  selector: 'app-add-dialog',
+  templateUrl: './add-dialog.component.html',
+  styleUrls: ['../book-dialog.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class AddDialogComponent {
+    action = FormActions.Add;
```

Comment:
Pls keep keep class interface secure:
Most of the fields have to be readonly or private with access by getter.